runtime/src/comm/ofi/comm-ofi.c

```diff
@@ -1830,15 +1917,73 @@ void chpl_comm_barrier(const char *msg) {
 
   DBG_PRINTF(DBG_BARRIER, "barrier '%s'", (msg == NULL) ? "" : msg);
 
-  if (numAmHandlersActive == 0) {
-    // Comm layer setup is not complete yet; use OOB barrier
-    chpl_comm_ofi_oob_barrier();
-  } else {
-    // Use OOB barrier for now, but we can do better in the future
+  if (pthread_equal(pthread_self(), pthread_that_inited)
```

Comment:
I don't quite understand this check, and it seems like this is wrong for fifo.

Why can't the main process do a regular barrier? I don't think this will hurt for qthreads, I just don't quit understand the limitation

Reply:
Ah, I see the motivation for this in the PR message, but I'm still not convinced this is right for fifo.

Reply:
It probably will break for fifo, but that doesn't concern me at the moment.  It not only doesn't hurt with Qthreads, it (or something else) is necessary in that config.  If we let the main process thread do RMA or an AM with Qthreads, it allocates a ptiTab[] entry and then we run out of those later because we only have as many of them as the comm concurrency (== numWorkers) plus 1 for the AM handler.  So this is a way to make ofi+qthreads work, at the expense of ofi+fifo failing.